custom_components/sleep_as_android/sensor.py

```diff
@@ -150,25 +181,6 @@ def name(self):
         """Return the name of the sensor."""
         return self._instance.create_entity_id(self._name)
 
-    @property
-    def state(self):
-        """Return the state of the entity."""
-        return self._state
-
-    @state.setter
-    def state(self, new_state: str):
-        """Set new state and fire events if needed.
-
-        Events will be fired if state changed and new state is not STATE_UNKNOWN.
-
-        :param new_state: str: new sensor state
-        """
-        if self._state != new_state:
-            self._state = new_state
-            self.async_write_ha_state()
-        else:
-            _LOGGER.debug("Will not update state because old state == new_state")
-
```

Comment:
I'm not sure about this part.
Sensor is calling `async_write_ha_state` manually now, but I could not remember why... Will check it...

Reply:
Could it be because polling is off?

Reply:
Possible. The Idea could be to not overload HA with same state.
If we removing this part of code, then we should enable `_attr_should_poll `

Reply:
Like even tho I am now a HA core member, I still haven't worked a lot with these kind of sensors and I think in our occasion, we push data. So I think the current approach would be the best. Altho I dislike the state setter. (I mean I really like the fact that I can turn on sleep tracking an my lights turn off immediatly, so having to wait max 30 seconds for a sensor update would be not needed)